src/components/Dashboard/EventCard.vue

```diff
@@ -57,17 +57,42 @@ const eventDateLabel = computed(() => {
 	}
 	const startDate = new Date(props.eventRoom.start * 1000)
 	const endDate = new Date(props.eventRoom.end * 1000)
-	const startDateString = startDate.toLocaleString(getCanonicalLocale(), { hour: '2-digit', minute: '2-digit' })
-	const endDateString = endDate.toLocaleString(getCanonicalLocale(), { hour: '2-digit', minute: '2-digit' })
-	const dateString = isToday.value
-		? t('spreed', 'Today')
-		: moment(startDate).calendar(null, {
-			nextDay: '[Tomorrow]',
-			nextWeek: 'dddd',
-			sameElse: 'dddd'
-		})
-	// FIXME should be a translated string
-	return `${dateString} ${startDateString} - ${endDateString}`
+	const isToday = startDate.toDateString() === new Date().toDateString()
+	const isTomorrow = startDate.toDateString() === new Date(Date.now() + 24 * 60 * 60 * 1000).toDateString()
+
+	let time
+	if (startDate.toDateString() === endDate.toDateString()) {
+		if (isToday || isTomorrow) {
+			// show the time only
+			const timeRange = Intl.DateTimeFormat(getCanonicalLocale(), {
+				hour: 'numeric',
+				minute: 'numeric',
+			}).formatRange(startDate, endDate)
+
+			// TRANSLATIORS: The time is shown in the format "HH:MM - HH:MM" e.g. "Tomorrow 10:00 - 11:00"
+			time = isToday
+				? t('spreed', 'Today {timeRange}', { timeRange })
```

Comment:
In general, I wouldn't introduce two new strings for this narrow usecase, but something more generic like:
`t('spreed', '{prefix} {dateTime}', { prefix, dateTime })`

See how it's done in d094cf5b21f47aa682d2f1372c8d0a0eaa4cf5b3, we can get 'today' and 'tomorrow' from Intl as well

Don't refactor everything, just make it less harder for translators and easier for us to later move logic to util